frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideOutAction.java

```diff
@@ -171,20 +173,23 @@ private void slideOutBranch(AnActionEvent anActionEvent, String branchName) {
     LOG.info("Sliding out '${branchName}' branch in memory");
     val newBranchLayout = branchLayout.slideOut(branchName);
 
-    try {
-      val macheteFilePath = gitRepository.getMacheteFilePath();
-      LOG.info("Writing new branch layout into ${macheteFilePath}");
-      branchLayoutWriter.write(macheteFilePath, newBranchLayout, /* backupOldLayout */ true);
-
-    } catch (BranchLayoutException e) {
-      val exceptionMessage = e.getMessage();
-      val errorMessage = "Error occurred while sliding out '${branchName}' branch" +
-          (exceptionMessage == null ? "" : ": " + exceptionMessage);
-      LOG.error(errorMessage);
-      VcsNotifier.getInstance(project).notifyError(/* displayId */ null,
-          getString("action.GitMachete.BaseSlideOutAction.notification.title.slide-out-fail.HTML").format(branchName),
-          exceptionMessage == null ? "" : exceptionMessage);
-    }
+    ApplicationManager.getApplication()
+        .invokeLater(() -> WriteAction
+            .compute(() -> Try.run(() -> {
+              val macheteFilePath = gitRepository.getMacheteFilePath();
+              LOG.info("Writing new branch layout into ${macheteFilePath}");
+              branchLayoutWriter.write(macheteFilePath, newBranchLayout, /* backupOldLayout */ true);
```

Comment:
Same here, TBD if `WriteAction` is needed currently. Albeit it will if #1303 ends up in a switch to a VFS write... so maybe better retain the `WriteAction`? or even better! @MaciejG604 pls first research #1303 and only then return to this PR 🔙